netmaster/master/network.go

```diff
@@ -338,18 +363,20 @@ func DeleteNetworkID(stateDriver core.StateDriver, netID string) error {
 		return err
 	}
 
-	// Check if there are any active endpoints
-	if hasActiveEndpoints(nwCfg) {
-		return core.Errorf("Error: Network has active endpoints")
-	}
-
-	if GetClusterMode() == "docker" {
+	if IsDNSEnabled() {
 		// detach Dns container
 		err = detachServiceContainer(nwCfg.Tenant, nwCfg.NetworkName)
 		if err != nil {
 			log.Errorf("Error detaching service container. Err: %v", err)
 		}
+	}
 
+	// Check if there are any active endpoints
+	if hasActiveEndpoints(nwCfg) {
```

Comment:
@DivyaVavili One slight view we might want to consider here. If active endpoints are present and we fail network delete, we might still want to have dns container lying around. as per the previous changes - in case of a failure Nothing would have changed. Now we will end up with partial cleanup..It might be better to have the ActiveEp check first. and fail it to start with.